src/views/Joseki/Joseki.tsx

```diff
@@ -184,6 +184,7 @@ export class Joseki extends React.Component<JosekiProps, any> {
         this.state = {
             move_string: "",             // This is used for making sure we know what the current move is. It is the display value also.
             current_node_id: undefined as string,    // The server's ID for this node, so we can uniquely identify it and create our own route for it,
+            most_recent_known_node: undefined as string,  // the value of current_node_id when the person clicked on a node not in the db
```

Comment:
Just a note: all these `as`s type annotations are getting lost because state, and therefore its members, is typed as `any`:

<img width="399" alt="Screen Shot 2021-12-09 at 6 55 21 PM" src="https://user-images.githubusercontent.com/25233703/145494136-81d01499-b2d9-455b-8e2f-845646add292.png">

If you want these annotations to be used by the compiler (and IDE tooling), the way to do that is to make an interface and pass it as a template arg to `React.Component`:

```
Interface JosekiState {
    current_move_id?: string;
    most_recent_known_node?: string;
    ...
}

React.Component<JosekiProps, JosekiState> {
    ...
```

Reply:
I so totally have to start doing that :D

Reply:
(although there's plenty of benefit in just recording that I intended it as a string :D :D )